src/printer.cpp

```diff
@@ -0,0 +1,314 @@
+#include <algorithm>
+
+#include "printer.h"
+
+namespace CSymPy {
+
+StrPrinter::StrPrinter(bool ordered) : BaseVisitor(this), order_(ordered) {
+
+}
+void StrPrinter::bvisit(const Basic &x) {
+    std::cout << "Basic" <<std::endl;
+}
```

Comment:
In the case of the StrPrinter, it will be invoked only if someone makes a new class.